lib/delayed/backend/base.rb

```diff
@@ -151,6 +147,17 @@ def fail!
         save!
       end
 
+      private
+
+      def display_name # rubocop:disable Metrics/AbcSize
+        @display_name ||= payload_object.job_data['job_class'] if payload_object.respond_to?(:job_data)
+        @display_name ||= payload_object.display_name if payload_object.respond_to?(:display_name)
+        @display_name ||= payload_object.class.name
+      rescue DeserializationError
```

Comment:
This `rescue` case should no longer apply once there are no reasons to recompute a display name after a job has already been enqueued. (Leaving it here for backwards-compat.)